parsl/dataflow/dflow.py

```diff
@@ -296,6 +297,109 @@ def std_spec_to_name(name, spec):
 
         return task_log_info
 
+    def _send_file_log_info(self, file: Union[File, DataFuture],
+                            task_record: TaskRecord, is_output: bool) -> None:
+        """ Generate a message for the monitoring db about a file. """
+        if self.monitoring and self.monitoring.capture_file_provenance:
+            file_log_info = self._create_file_log_info(file, task_record)
+            # make sure the task_id is None for inputs
+            if not is_output:
+                file_log_info['task_id'] = None
+            self.monitoring.send((MessageType.FILE_INFO, file_log_info))
+
+    def _create_file_log_info(self, file: Union[File, DataFuture],
+                              task_record: TaskRecord) -> Dict[str, Any]:
+        """
+        Create the dictionary that will be included in the ,omitoring db.
+        """
+        file_log_info = {'file_name': file.filename,
+                         'file_id': str(file.uuid),
+                         'run_id': self.run_id,
+                         'task_id': task_record['id'],
+                         'try_id': task_record['try_id'],
+                         'timestamp': file.timestamp,
+                         'size': file.size,
+                         'md5sum': file.md5sum
+                         }
+        return file_log_info
+
+    def register_as_input(self, f: Union[File, DataFuture],
+                          task_record: TaskRecord):
+        """ Register a file as an input to a task. """
+        if self.monitoring and self.monitoring.capture_file_provenance:
+            self._send_file_log_info(f, task_record, False)
+            file_input_info = self._create_file_io_info(f, task_record)
+            self.monitoring.send((MessageType.INPUT_FILE, file_input_info))
+
+    def register_as_output(self, f: Union[File, DataFuture],
+                           task_record: TaskRecord):
+        """ Register a file as an output of a task. """
+        if self.monitoring and self.monitoring.capture_file_provenance:
+            self._send_file_log_info(f, task_record, True)
+            file_output_info = self._create_file_io_info(f, task_record)
+            self.monitoring.send((MessageType.OUTPUT_FILE, file_output_info))
+
+    def _create_file_io_info(self, file: Union[File, DataFuture],
+                             task_record: TaskRecord) -> Dict[str, Any]:
+        """
+        Create the dictionary that will be included in the monitoring db
+        """
+        file_io_info = {'file_id': str(file.uuid),
+                        'run_id': self.run_id,
+                        'task_id': task_record['id'],
+                        'try_id': task_record['try_id'],
+                        }
+        return file_io_info
+
+    def _register_env(self, environ: ParslExecutor) -> None:
```

Comment:
This environment behaviour is orthogonal to provenance, I think, and is interesting for non-provenance users too - you could make it as a separate PR and make it happen for every task when monitoring is turned on.

If this is intended only for human consumption, you might consider sending a single value, the `repr` of the executor, which will capture in fairly human readable form all of the paramters of the executor. (executors should be expected to give a sensible `repr` because of the expectation that we can log a `parsl.Config` object at the start of a workflow run)

A downside to that is you don't get separated out fields for provider label/worker_init/launcher in machine readable form - but I think you should need to justify why those three fields are specifically interesting vs the whole `repr`.